src/linker/Linker.Steps/MarkStep.cs

```diff
@@ -529,7 +529,8 @@ void ProcessDefaultImplementation (TypeDefinition typeWithDefaultImplementedInte
 			if (!Annotations.IsInstantiated (typeWithDefaultImplementedInterfaceMethod))
 				return;
 
-			MarkInterfaceImplementation (implementation, typeWithDefaultImplementedInterfaceMethod);
+			if (!Annotations.IsMarked (implementation))
```

Comment:
Should we move this check into the `MarkInterfaceImplementation`? There are multiple callsites for that method - I guess none has this problem... but it would not hurt and would make the code more robust.